src/sentry/static/sentry/app/views/substances/substances.jsx

```diff
@@ -70,19 +78,66 @@ class Substances extends React.Component {
     ];
   }
 
+  selectGroupBy(selection) {
+    this.setState({groupBy: {value: selection.value, label: selection.label}});
+  }
+
+  renderDropdown() {
+    const groupedItems = [
+      {
+        value: 'Fields',
+        label: <div>Field</div>,
+        items: [
+          {
+            value: 'sample',
+            label: 'Sample',
+          },
+          {
+            value: 'container',
+            label: 'Container',
+          },
+          {
+            value: 'sample_type',
+            label: 'Sample Type',
+          },
+        ],
+      },
+    ];
+
+    return (
+      <DropdownAutoComplete
+        items={groupedItems}
+        alignMenu="left"
+        onSelect={this.selectGroupBy.bind(this)}
+      >
+        {({isOpen, selectedItem}) => (
```

Comment:
It's interesting that they use this pattern, i.e. that there is a function for rendering each button.